src/containers/FormikForm/GrepCodesField.tsx

```diff
@@ -6,20 +6,207 @@
+          code: grepCode,
+          title: `${grepCode} - ${grepCodeTitle}`,
+        });
+      } else if (!isGrepCodeSaved) {
+        setTimeout(() => {
+          helpers.setError(`${t("errorMessage.grepCodes")}${grepCode}`);
+        }, 0);
+      }
+    }
+    return newGrepCodeNames;
+  };
+
+  const updateGrepCodes = async (values: string[]) => {
+    helpers.setError(undefined);
+    const trimmedValues = uniq(values.map((v) => v.toUpperCase().trim()));
+    // Grep code is added
+    if (trimmedValues.length > field.value.length) {
+      const addedGrepCodes = difference(trimmedValues, field.value);
+      const grepCodesWithNames = await fetchGrepCodeTitles(addedGrepCodes);
+      const grepCodesObject = grepCodesWithNames.reduce((acc, c) => ({ ...acc, [c.code]: c.title }), grepCodes);
+      setGrepCodes(grepCodesObject);
+      helpers.setValue(Object.keys(grepCodesObject));
+      return;
+    }
+    // Grep code is removed
+    if (trimmedValues.length < field.value.length) {
+      const filteredGrepCodes = trimmedValues.reduce(
+        (acc, c) => (values.includes(c) ? { ...acc, [c]: grepCodes[c] } : acc),
+        {},
+      );
+      setGrepCodes(filteredGrepCodes);
+      helpers.setValue(values);
+      return;
+    }
+  };
+
+  const collection = useMemo(() => {
+    return createListCollection({
+      items: searchQuery.data?.results ?? [],
+      itemToString: (item) => item.title,
+      itemToValue: (item) => item.code,
+      isItemDisabled: (item) => !!grepCodes[item.code],
+    });
+  }, [grepCodes, searchQuery.data?.results]);
+
   return (
-    <>
-      <FormikField name="grepCodes" label={t("form.grepCodes.label")}>
-        {({ field, form }: FieldProps<string[], FormikValues>) => <GrepCodesFieldContent field={field} form={form} />}
-      </FormikField>
-    </>
+    <FormField name="grepCodes">
+      {({ field, meta }) => (
+        <FieldRoot>
+          <FieldLabel>{t("form.grepCodes.label")}</FieldLabel>
+          <FieldHelper>{t("form.grepCodes.description")}</FieldHelper>
+          <Text color="text.error" aria-live="polite">
+            {meta.error}
+          </Text>
+          <ComboboxRoot
+            collection={collection}
+            translations={translations}
+            onInputValueChange={(details) => setQuery(details.inputValue)}
+            inputValue={query}
+            onValueChange={(details) => updateGrepCodes(details.value)}
+            value={field.value}
+            multiple
+            positioning={{ strategy: "fixed" }}
+            variant="complex"
+            context="standalone"
+          >
+            <GenericComboboxInput
+              placeholder={t("form.grepCodes.placeholder")}
+              isFetching={searchQuery.isFetching}
+              onKeyDown={(event) => {
+                if (event.key === "Enter" && !!query.trim()) {
+                  updateGrepCodes([...field.value, query]);
+                }
+              }}
```

Comment:
Hva er tanken med denne? At man automatisk skal legge til grepkoder på enter? Hva om den er ugyldig?

Reply:
Å kunne legge til nye koder ved enter ja, det er validering av input i updateGrepCodes, ugyldige koder blir ikke lagt til

Reply:
Gir det ikke da mer mening å kun tillate grep-koder som dukker opp i søkeresultat? Eller blir det dumt hvis API'et går ned?

Reply:
Det ble gjort sånn for å beholde samme oppførsel som har vært tidligere, er det noe vi kan endre på @gunnarvelle ?

Reply:
Det ble gjort sånn for å beholde samme oppførsel som tidligere, er det noe vi kan endre på @gunnarvelle ?

Reply:
Søket er vårt eget søk på koder som allerede er lagt til i artikler. Dersom vi aldri har lagt inn en gitt kode før må jo det gå an å legge til en ny som ikkje dukker opp i søket.